packages/bootstrap-chat/README.md

```diff
@@ -0,0 +1,157 @@
+ll still have the look and feel of the original chat message component, however with a custom message property.
+
+
+## The `<ChatApp />` component
+
+The `ChatApp` component exposes the following props: 
+
+*Note: All render-based props will receive as part of their args both `mephistoContext` and `appContext`, which will be described in a separate section below.*
+
+
+### `renderMessage({ message, idx, mephistoContext, appContext })`
+A render prop that if implemented, overides the way a chat message is rendered.
+
+**Arguments:**
+
+**`message`**: An object with the following schema, this comes from the ParlAI framework:
+
+```
+{
+    'id', // the speaker id for the message
+    'message_id', // a unique id for the current message
+    'text', // the actual message text, the message should not be displayed if this value is undefined or an empty string
+    'task_data', // any additional data payload, also used to update the `taskContext`
+}
+```
+
+**`idx`**: The 0-based index of the current message in the list of all messages
+
+
+### `renderSidePane({ mephistoContext, appContext })`
+A render prop that if implemented, overrides the way the side pane is rendered.
+
+### `renderTextResponse({ onMessageSend, inputMode, active, mephistoContext, appContext })`
+A render prop that if implemented, overrides the way the response text box is rendered. The default implementation of `ChatApp` is rendering the `<TextResponse />` component here.
+
+Note: If `renderResponse()` is also implemented, this render prop will be ignored.
+
+This render prop is for cases where you'd like to minimally alter the response text box, while still keeping the default UI states for things like handling disconnects, inactivity, and the done state. For larger customizations and more control, we recommend implementing `renderReponse()` instead.
+
+**Arguments:**
+
+**`onMessageSend`**: A callback provided for you to invoke in your implementation. Invoking this callback will send the passed in message back to the server. Example usage: `onMessageSend({ text: textValue, task_data: {} })`
+
+**`inputMode`**: Represents an enumeration for the status of the chat task. It is set to one of the following:
+
+- `WAITING`: The current agent is ready but is waiting for a response from the server, e.g. waiting for an incoming chat message.
+- `INACTIVE`: Occurs when something goes wrong, usually if the agent experiences a disconnect, timeout, the task is returned, or has expired.
+- `DONE`: The task is complete.
+- `READY_FOR_INPUT`: The current agent is ready and is ready to send over data.
+- `IDLE`: (TODO: This doesn't seem to be set anywhere. Should we remove?)
```

Comment:
So we have an `IDLE` state for the `INPUT_MODE` enum, however I don't see usage anywhere. The default implementation doesn't ever set the state to `IDLE` for any case. Thoughts on whether to keep this enum or delete?

Reply:
Yeah we can kill `IDLE`, I don't think we need it anymore.